depthai_bridge/include/depthai_bridge/SpatialDetection2DConverter.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <deque>
+#include <memory>
+#include <string>
+
+#include <rclcpp/time.hpp>
+#include <vision_msgs/msg/detection2_d_array.hpp>
+#include "depthai/pipeline/datatype/SpatialImgDetections.hpp"
+
+namespace dai {
+
+namespace ros {
+
+namespace VisionMsgs = vision_msgs::msg;
+using SpatialDetection2DArrayPtr = VisionMsgs::Detection2DArray;
+
+class SpatialDetection2DConverter {
```

Comment:
What is the exact use case here? Current SpatialDetectionConverter has [a method](https://github.com/luxonis/depthai-ros/blob/humble/depthai_bridge/src/SpatialDetectionConverter.cpp#L95) to output vision_msgs format

Reply:
Yes, that is true, but I think Detection3DArray is not necessary, Detection2DArray is enough. And actually you are not making use of it [here](https://github.com/luxonis/depthai-ros/blob/10930361f8d44725b656a1c0b33f5e2975f550b6/depthai_bridge/src/SpatialDetectionConverter.cpp#L139).
But maybe I can add another function to the exitent SpatialDetectionConverter to publish Detection2DArray, let me know your opinion on this.

Reply:
@Serafadam what should I do in this case?

Reply:
Hi, if it really needed then I would update spatial detection converter to have an option to choose which detection array to publish, with default being detection3d array.